src/main/java/producer/RecordBatch.java

```diff
@@ -0,0 +1,62 @@
+package producer;
+
+import org.tinylog.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordBatch {
+    private final int maxBatchSizeInBytes;
+    private int currBatchSize;
+    private int numRecords;
+    private List<byte[]> batch;
+
+    public RecordBatch() {
+        this(10_000); // default batch size: 10 KB = 100,00 bytes
```

Comment:
Could be nice to have this as a constant

Reply:
right I was trying to do that but it kept giving an error earlier, retried it just now and it worked when i included the `static` keyword

Reply:
Nitpick: I actually think this should be `10_240` because technically 10 KB = 10 × 1024 bytes = 10,240 bytes
```java
this(10_240); 
```

Reply:
>right I was trying to do that but it kept giving an error earlier, retried it just now and it worked when i included the static keyword

You can make it a private static long

Reply:
i decided to change all byte variables from long --> int cuz after thinking about it, our system will never need to handle any data batches bigger than an int which is equiv to ~2 GB